esp-hal/src/rtc_cntl/sleep/mod.rs

```diff
@@ -131,6 +131,71 @@ impl<'a, 'b> RtcioWakeupSource<'a, 'b> {
     }
 }
 
+macro_rules! uart_wakeup_impl {
```

Comment:
could have implemented some trait on the UARTs and have generics but I think in this very simple case a macro is not a bad way to do it